src/intrinsics/ecma262/Object.js

```diff
@@ -60,6 +60,8 @@ export default function(realm: Realm): NativeFunctionValue {
     // 1. Let to be ? ToObject(target).
     let to = To.ToObjectPartial(realm, target);
```

Comment:
Just do:

```js
if (to.isSimpleObject()) {
  to.makeSimple();
}
```

This seems weird. Like why mark something as simple if it already is? Well, it is because the first read is derived from a bunch of considerations. One of those considerations is whether it is partial.

Later it becomes partial and that switches the default to false. So if we set it explicitly in the beginning, then it'll remain partial.

cc @hermanventer This seems a bit odd in the model that isSimpleObject switches property like this when it becomes partial.

Reply:
Hm. I guess there is no way to make an object not simple again after it has once been explicitly declared simple. So maybe this is not such a good idea.

The unit test is an example of this when a Object.defineProperty happens after the assign.

Reply:
https://github.com/facebook/prepack/pull/1456